### DISCUSSION TITLE
How is this different from @cloudinary/react

### DISCUSSION THREAD

**Discussion by marcobiedermann** [cloudinary-community/next-cloudinary#296] [Q&A] [answered]
Looking at your official documentation, there is also an SDK for React. What is the difference between `@cloudinary/react` and this package?

**Comment by colbyfayock** [comment, marked as answer]
hey @marcobiedermann great question!

this library takes advantage of Next.js-specific tech, here are a few examples

the CldImage component is a wrapper around Next Image, which is a fantastic image component. we extend that using Cloudinary tech both for optimization and being able to apply transformations on top. with the React SDK, you don't get the image-specific features that Next Image provides, but you get the other Cloudinary-specific features, which may be more in depth than what is currently available here. they both also provide a different Developer Experience, so you may prefer one over the other from the usability perspective.

The React SDK doesn't the Upload Widget, this provides a drop-in component

The React SDK includes an AdvancedVideo component, but this provides the Cloudinary Video Player as a component via CldVideoPlayer. both capable of using the Cloudinary SDKs and tech, but the Cloudinary Video Player provides more enhanced features for the user experience

overall, this library is intended to provide first class support for Next.js applications, where the React SDK is broad React support, so the granularity can help in some ways with this library

let me know if you have any specific questions

**Reply by MorningLit** [reply]
Is it possible for the CldImage component to entirely replace the Next Image component? One of the reasons is because I'm trying to find alternatives to Next.js's Image component because I've hit the limit for their image optimisations (1000 images), and if my understanding is correct, this CldImage still uses Next.js's Image component under the hood and thus still consumes Next.js's image optimisations?

**Reply by colbyfayock** [reply]
yup! CldImage wraps Next Image so it's a drop-in replacement with the caveat that you need to reference Cloudinary images

```
<CldImage src="<Cloudinary Public ID>" ... />
```

the technical details are that it uses the Next Image loader prop to pass in a custom function allowing us to reproduce everything the Image component does, but using Cloudinary URLs. this also allows us to extend it with our own custom props for things like transformations

**Comment by marcobiedermann** [comment]
@colbyfayock 

Thank you very much for your quick and detailed explanation 👍 
Yeah, I was just wondering after I participated in your workshop yesterday. Great job by the way 💪

**Reply by colbyfayock** [reply]
thank you! 😁
